rules/cross-platform/multiple_alerts_edr_elastic_defend_by_host.toml

```diff
@@ -0,0 +1,97 @@
+[metadata]
+creation_date = "2025/11/19"
+maturity = "production"
+updated_date = "2025/11/19"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule uses alert data to determine when multiple alerts from Elastic Defend involving the same host are triggered.
+Analysts can use this to prioritize triage and response, as these hosts are more likely to be compromised.
+"""
+from = "now-60m"
+interval = "30m"
+language = "esql"
+license = "Elastic License v2"
+name = "Multiple Elastic Defend Alerts by Agent"
+risk_score = 73
+rule_id = "ab25369e-ea5e-46f1-9cd5-478a0a4a131a"
+severity = "high"
+tags = ["Use Case: Threat Detection", "Rule Type: Higher-Order Rule", "Resources: Investigation Guide", "Data Source: Elastic Defend"]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-endpoint.alerts-* metadata _id
+| eval target_time_window = DATE_TRUNC(24 hours, @timestamp)
+| where event.code in ("malicious_file", "memory_signature", "shellcode_thread", "behavior") and
+        agent.id is not null and not rule.name in ("Multi.EICAR.Not-a-virus")
+| stats Esql.alerts_count = COUNT(*),
+        Esql.event_code_distinct_count = count_distinct(event.code),
+        Esql.rule_name_distinct_count = COUNT_DISTINCT(rule.name),
+        Esql.file_hash_distinct_count = COUNT_DISTINCT(file.hash.sha256),
+        Esql.process_name_distinct_count = COUNT_DISTINCT(process.entity_id),
+        Esql.event_code_values = VALUES(event.code),
+        Esql.rule_name_values = VALUES(rule.name),
+        Esql.message_values = VALUES(message),
+        Esql.file_path_values = VALUES(file.path),
+        Esql.dll_path_values = VALUES(dll.path),
+        Esql.process_executable_values = VALUES(process.executable),
+        Esql.process_parent_executable_values = VALUES(process.parent.executable),
+        Esql.process_command_line_values = VALUES(process.command_line),
+        Esql.process_hash_sha256_values = VALUES(process.hash.sha256),
+        Esql.file_hash_sha256_values = VALUES(file.hash.sha256),
+        Esql.dll_hash_sha256_values = VALUES(dll.hash.sha256) by agent.id
```

Comment:
I think we need to test whether this could cause performance problems, coz it looks expensive 👀